docs/design/proposals/decorators.md

```diff
@@ -0,0 +1,265 @@
+# KREP-003 Decorators (Collection Watching)
+
+## Summary
+
+KREP-003 introduces Decorators (a.k.a Collection Watching), an extension of
+[Collections](https://github.com/kubernetes-sigs/kro/pull/679) and
+[ExternalRefs](https://kro.run/docs/concepts/resource-group-definitions#using-externalref-to-reference-objects-outside-the-resourcegraphdefinition).
+We extend `externalRef` to support watching a collection of objects, rather than just a single
+object. Additionally, this KREP also recommends renaming `externalRef` to `watch` in a future API
+version of KRO.
+
+## Motivation
+
+The Kubernetes’ operational model was designed with two tiers of responsibility: cluster
+administration and application development. Cluster administrators pre-configure globally scoped
+APIs for compute (NodePools), storage (StorageClass), networking (GatewayClass), permissions (RBAC).
+This model extends to KRO, where cluster administrators configure RGDs that provide higher-order
+interfaces that abstract application definition. Together, these policies enforce a set of
+guardrails for application development. Once they are in place, the cluster is considered
+“application ready”, and application developers can begin to apply namespace scoped APIs to deploy
+their applications. In practice, these guardrails evolve over time, but the mental model is
+sufficient for the purposes of this design.
+
+While KRO enables cluster administrators to create RGDs that define new APIs to enforce application
+guardrails, it requires application developers to “buy-in” and begin using them. This can be
+challenging in large multi-tenant environments that contain large numbers of application teams that
+leverage existing and heterogeneous tooling. Further, all policies must be bundled together into a
+single monolithic RGD. This requires cluster administrators to centralize all application guardrails
+into a single monolithic configuration. Decorators provide a solution to these problems.
+
+## Proposed API and Behavior Changes
+
+1. Introduce `Selector` and `NamespaceSelector` to `ExternalRef`
```

Comment:
when using Selector and NamespaceSelector, there is even more reason to keep it as Reference because even Kubernetes calls it reference